component/portal/src/main/java/org/exoplatform/portal/mop/storage/cache/CacheSiteStorage.java

```diff
@@ -79,17 +91,22 @@ public void remove(PortalData config) {
     try {
       super.remove(config);
     } finally {
-      siteFutureCache.remove(getSiteKey(config.getKey()));
+      clearCache();
     }
   }
 
   @Override
   public PortalData getPortalConfig(SiteKey key) {
-    PortalData portalData = siteFutureCache.get(null, key);
+    PortalData portalData = (PortalData) siteFutureCache.get(null, key);
     return portalData == null || portalData.isNull() ? null : portalData;
   }
 
-  private SiteKey getSiteKey(PortalKey portalKey) {
-    return new SiteKey(portalKey.getType(), portalKey.getId());
+  @Override
+  public List<PortalData> getSites(SiteFilter siteFilter) {
+    return  (List<PortalData>) siteFutureCache.get(siteFilter.hashCode(), siteFilter);
```

Comment:
For cache efficiency, instead of caching the list of `PortalData` for each `Filter` you can cache only corresponding site names. (a very reduced cache size)
Once the site names are cached, you can use the `site cache` to retrieve the sites one by one:

```java
  // to add in ________SiteStorageImpl________
  public List<PortalData> getSites(SiteFilter siteFilter) {
    return getSiteKeys(siteFilter).map(this::getPortalConfig).toList();
  }

  public List<PortalData> getSiteKeys(SiteFilter siteFilter) {
    return siteDAO.findSiteKeys(siteFilter);
  }
```

```java
  // to add in ________CacheSiteStorage________
  public List<PortalData> getSiteKeys(SiteFilter siteFilter) {
    return siteKeysByFilterFutureCache.get(null, siteFilter);
  }
```

This way, we will have a cache for `PortalConfig` by `SiteKey` and a `List` of `SiteKey` (**only**) by `SiteFilter` which will improve considerably the `cache lifecycle` management and reduces the **memory size consumption**.